src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -122,12 +145,58 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
             } catch (Exception e) {
                 logger.atError().cause(e).kv("thingName", thing.getThingName())
                         .kv("certificateId", certificate.getCertificateId())
-                        .log("Failed to verify certificate thing association. Check that the core device's IoT policy"
-                                + " grants the greengrass:VerifyClientDeviceIoTCertificateAssociation permission.");
+                        .log("Failed to verify certificate thing association. Check that the core device's "
+                                + "IoT policy grants the greengrass:VerifyClientDeviceIoTCertificateAssociation "
+                                + "permission.");
                 throw new CloudServiceInteractionException(
                         String.format("Failed to verify certificate %s thing %s association",
                                 certificate.getCertificateId(), thing.getThingName()), e);
             }
         }
+
+        @Override
+        public Stream<Thing> getThingsAssociatedWithCoreDevice() {
+            DeviceConfiguration configuration = clientFactory.getDeviceConfiguration();
+            String thingName = Coerce.toString(configuration.getThingName());
+
+            ListClientDevicesAssociatedWithCoreDeviceRequest request =
+                    ListClientDevicesAssociatedWithCoreDeviceRequest.builder()
+                            .coreDeviceThingName(thingName)
+                            .build();
+
+            try (GreengrassV2Client client = getGGV2Client(configuration)) {
+                ListClientDevicesAssociatedWithCoreDeviceResponse response =
+                        client.listClientDevicesAssociatedWithCoreDevice(request);
+
+                return response.associatedClientDevices().stream()
+                        .map(AssociatedClientDevice::thingName)
+                        .map(Thing::of);
+            }  catch (SdkClientException | AwsServiceException e) {
```

Comment:
what about other exceptions? They're just going to bubble up. Safer to just catch all exceptions here?